EventBus.Kafka/KafkaEventSubscriber.cs

```diff
@@ -0,0 +1,107 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Finaps.EventBus.Kafka.Configuration;
+using Finaps.EventBus.Core;
+using Finaps.EventBus.Core.Abstractions;
+using Finaps.EventBus.Core.Utilities;
+using Microsoft.Extensions.Logging;
+using Confluent.Kafka;
+using System.Text;
+using System.Collections.Generic;
+
+namespace Finaps.EventBus.Kafka
+{
+
+  internal class KafkaEventSubscriber : IEventSubscriber
+  {
+    private bool _disposed;
+    private readonly ILogger<KafkaEventSubscriber> _logger;
+    public event AsyncEventHandler<IntegrationEventReceivedArgs>? OnEventReceived;
+    private KafkaOptions _options;
+    private IConsumer<Ignore, string> _consumer;
+
+    internal KafkaEventSubscriber(
+      ILogger<KafkaEventSubscriber> logger,
+      KafkaOptions options
+    )
+    {
+      _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+      _options = options;
+      _consumer = new ConsumerBuilder<Ignore, string>(
+        new ConsumerConfig
+        {
+          GroupId = options.GroupId,
+          BootstrapServers = options.Brokers,
+          AutoOffsetReset = AutoOffsetReset.Earliest,
+        }
+      ).Build();
+
+      //TODO: Make sure all topics exist, error if trying to subscribe to topic thats not
+      _consumer.Subscribe(options.TopicNames);
```

Comment:
Would it make sense to move this to the `SubscribeAsync`?

Reply:
Not really since the core eventbus is setup like this:

`      foreach (var eventName in _subscriptionsManager.GetSubscriptions())
      {
        await _subscriber.SubscribeAsync(eventName);
      }
`

Reply:
So then it would subscribe to the topics for each event subscription